src/containers/RecordingsTable.tsx

```diff
@@ -54,40 +52,32 @@ const RecordingsTable: FunctionComponent<Props> = ({ recordings, sortings, onDel
 
     recordings = sortByKey(recordings, 'recordingLabel');
 
-    const effect = async () => {
-        recordings.forEach(rec => {
-            (async () => {
-                if ((!rec.recordingInfo) && (!rec.fetchingRecordingInfo)) {
-                    // todo: use calculationPool for this
-                    rec.fetchingRecordingInfo = true;
-                    try {
-                        const info = await getRecordingInfo({recordingObject: rec.recordingObject, hither});
-                        onSetRecordingInfo({ recordingId: rec.recordingId, recordingInfo: info });
-                    }
-                    catch (err) {
-                        console.error(err);
-                        return;
-                    }
-                }
-            })();
-        });
-    }
-    useEffect(() => { effect() })
+    const handleViewRecording = useCallback((recording: Recording) => {
+        workspaceRouteDispatch({
```

Comment:
The routing is now handled by workspaceRouteDispatch mechanism, which is way better than forming the path. For example, this handles when the user clicks on a recording and takes us to the RecordingView page. It modifies the site URL using a virtual route. Will be described further below.

Reply:
Question: would it suffice to have one action (like 'navigate' or something?), but let the existence of recording ID/sorting ID determine what the target page ought to be?

Not saying we should do it that way, just wondering if that would also work